python/cudf/cudf/utils/dtypes.py

```diff
@@ -161,14 +163,18 @@ def to_cudf_compatible_scalar(val, dtype=None):
     if val is None:
         return val
 
-    dtype = "str" if is_string_dtype(dtype) else dtype
-
     if not is_scalar(val):
         raise ValueError(
             f"Cannot convert value of type {type(val).__name__} "
             " to cudf scalar"
         )
 
+    if isinstance(val, (np.ndarray, cp.ndarray)) and val.ndim == 0:
+        val = val.item()
```

Comment:
Future optimization: May want to allow 0-dim cupy arrays to pass through here and to use as the underlying memory for a device scalar.

Reply:
Noted, I suppose the only hurdle would be str types which cupy doesn't support. While we would have to specially handle